apps/web/pages/signup.tsx

```diff
@@ -34,43 +43,130 @@ type FormValues = z.infer<typeof signupSchema>;
 
 type SignupProps = inferSSRProps<typeof getServerSideProps>;
 
-const checkValidEmail = (email: string) => z.string().email().safeParse(email).success;
+const FEATURES = [
+  {
+    title: "connect_all_calendars",
+    description: "connect_all_calendars_description",
+    icon: CalendarHeart,
+  },
+  {
+    title: "workflow_automation",
+    description: "workflow_automation_description",
+    icon: Hexagon,
+  },
+  {
+    title: "scheduling_for_your_team",
+    description: "scheduling_for_your_team_description",
+    icon: Clock,
+  },
+];
+
+function UsernameField({
```

Comment:
Use a specific usename field and debounce the value to not call this on every key press like the zod schema approach is doing in website

Reply:
This always considers premium logic, we should disable that if it is a self hosted.

Reply:
We are using this already:
```
export const getUsernameAvailabilityComponent = (isPremium: boolean) => {
  if (isPremium)
    return dynamic(() => import("./PremiumTextfield").then((m) => m.PremiumTextfield), { ssr: false });
  return dynamic(() => import("./UsernameTextfield").then((m) => m.UsernameTextfield), { ssr: false });
};
```